pkg/generate/app/imageref.go

```diff
@@ -104,7 +104,10 @@ func (g *imageRefGenerator) FromStream(stream *imageapi.ImageStream, tag string)
 		return nil, err
 	}
 
-	if len(tag) == 0 && len(ref.Tag) == 0 {
+	switch {
+	case len(tag) > 0:
+		ref.Tag = tag
```

Comment:
@kargakis that's what we had before... basically you didn't specify a tag and no tag is associated with the docker image reference of the image stream